tool/src/massive/munit/Config.hx

```diff
@@ -54,9 +57,103 @@ class Config
 		this.dir = dir;
 		this.currentVersion = currentVersion;
 		targetTypes = defaultTargetTypes;
-		configFile = dir.resolveFile(".munit");
-		exists = configFile.exists;
-		if(exists) load();
+		if(!parseCommandLineConfig())
+		{
+			configFile = dir.resolveFile(".munit");
+			exists = configFile.exists;
+			if(exists) load();
+		}
+	}
+
+	private function parseCommandLineConfig():Bool
+	{
+		var args = Sys.args();
+		if(args[1] != CLI_CONFIG)
+			return false;
+
+		src = File.create(args[2], dir);
+		bin = File.create(args[3], dir, true);
+		report = File.create(args[4], dir, true);
+		hxml = null;
+
+		configVersion = getCommandLineConfigValue("version");
+
+		var resources = getCommandLineConfigValue("resources");
+		this.resources = resources != null ? File.create(resources, dir) : null;
+
+		var templates = getCommandLineConfigValue("templates");
+		this.templates = templates != null ? File.create(templates, dir) : null;
+
+		var classPaths = getCommandLineConfigValues("classPath");
+		this.classPaths = [];
+		if(classPaths != null)
+			for(classPath in classPaths)
+				this.classPaths.push(File.create(classPath, dir));
+
+		coveragePackages = getCommandLineConfigValues("coveragePackage");
+		coverageIgnoredClasses = getCommandLineConfigValues("coverageIgnoredClass");
+
+		targets.push(getCommandLineTarget());
+		return true;
+	}
+
+	private function getCommandLineConfigValue(name:String):String
+	{
+		var result = getCommandLineConfigValues(name);
+		return result == null ? null : result[0];
+	}
+
+	private function getCommandLineConfigValues(name:String):Array<String>
+	{
+		var args = Sys.args();
+		var result:Array<String> = [];
+		var length = args.length;
+		for(i in 0...length)
+		{
+			if(args[i] == ("config:" + name) && length > i)
```

Comment:
`length > i` is always `true`

Reply:
@dobrusev this we need to fix to `length > (i + 1)` or `(i + 1) < length` so we can access `args[i + 1]`

Reply:
or `for(i in 0...length-1)`

Reply:
yep, even better